### ISSUE TITLE
Assign Random ports and Handle Connect/Disconnect Events

### ISSUE THREAD

**Issue by v1ct0rv** [jpillora/chisel#410]
Hi,

I'm working on a project to expose rtsp feed from remote IP cameras, The idea is the devices (DVR) use the chisel client and expose the rtsp feed and ssh port using reverse port forwarding, something like `chisel client -v remote-server.com:9312 R:8081:localhost:554`, then we can access the camera feed using `rtsp://username:password@remote-server.com:8081`, I have some questions:

1. It is possible to assign a random port on the server? the devices will does not know what port are used, so, would be nice if the serve can assign a random port (from a port range if its possible) 
2. We need to know which devices are connected, which port is assigned to them and is mapping to what port, we also want to know when a device is disconnected, there is a way to subscribe to events like connected, disconnected, heartbeats?

Thanks in Advance!

**Comment by jpillora** [comment]
Fairly advanced use case, I’d suggest write your own Go program to
implement this logic - and import chisel as a package

**Comment by v1ct0rv** [comment]
Got it, I will try.

Do you have metrics about chisel? how many clients supports a server, resource consumption, etc?

**Comment by cedricve** [comment]
Agree, we'll do the same for the kerberos.io project. https://github.com/kerberos-io
We will ship a custom version of chisel on a hardware device that does camera discovery and tunnel them through chisel to the other side where we fetch the streams and so some processing.
